airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/fix/java/commands.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+import logging
+import sys
+from typing import Optional
+
+import asyncclick as click
+import dagger
+from pipelines.airbyte_ci.format.consts import DEFAULT_FORMAT_IGNORE_LIST
+from pipelines.cli.click_decorators import click_ignore_unused_kwargs
+from pipelines.helpers.utils import sh_dash_c
+from pipelines.models.contexts.click_pipeline_context import ClickPipelineContext, pass_pipeline_context
+
+
+@click.command()
+@pass_pipeline_context
+@click_ignore_unused_kwargs
+async def java(ctx: ClickPipelineContext):
+    """Format java, groovy, and sql code via spotless."""
+    dagger_client = ctx.params["dagger_client"]
+
+    format_container = await (
+        dagger_client.container()
+        .from_("openjdk:17.0.1-jdk-slim")
+        .with_exec(
+            sh_dash_c(
+                [
+                    "apt-get update",
+                    "apt-get install -y bash",
+                ]
+            )
+        )
+        .with_mounted_directory(
+            "/src",
+            dagger_client.host().directory(
+                ".",
+                include=[
+                    "**/*.java",
+                    "**/*.sql",
+                    "**/*.gradle",
+                    "gradlew",
+                    "gradlew.bat",
+                    "gradle",
+                    "**/deps.toml",
+                    "**/gradle.properties",
+                    "**/version.properties",
+                    "tools/gradle/codestyle/java-google-style.xml",
+                    "tools/gradle/codestyle/sql-dbeaver.properties",
+                ],
+                exclude=DEFAULT_FORMAT_IGNORE_LIST,
+            ),
+        )
+        .with_workdir(f"/src")
+        .with_exec(["./gradlew", "spotlessApply"])
```

Comment:
Piggy-backing on one of @alafanechere 's comments, you really want to base this container on https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/steps/gradle.py#L150

This can totally be a follow-up task, in which case please file an issue.

Reply:
I'm sold - but I'd definitely like a follow up as I think I need direct collaboration with one of you to get this done safely 😅